src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py

```diff
@@ -85,6 +88,188 @@
+                image = image.transpose(0, 3, 1, 2)
+                image = torch.from_numpy(image)
+            elif isinstance(image[0], torch.Tensor):
+                image = torch.cat(image, dim=0)
+
+        image_batch_size = image.shape[0]
+
+        if image_batch_size == 1:
+            repeat_by = batch_size
+        else:
+            # image batch size is the same as prompt batch size
+            repeat_by = num_images_per_prompt
+
+        image = image.repeat_interleave(repeat_by, dim=0)
+
+        image = image.to(device=device, dtype=dtype)
+
+        return image
+
+    def prepare_image(
+        self, width, height, batch_size, num_images_per_prompt, device, dtype, do_classifier_free_guidance
+    ):
+        image = self._prepare_image(
+            self.image_original, width, height, batch_size, num_images_per_prompt, device, dtype
+        )
+        if do_classifier_free_guidance:
+            image = torch.cat([image] * 2)
+        self.image = image  # .clone()
+
+
+# Multi controlnet draft #2621
+
+
+class MultiControlNet(nn.Module):
+    def __init__(self, controlnets: List[ControlNetModel]):
+        super().__init__()
+        self.nets = nn.ModuleList(controlnets)
+
+    @property
+    def device(self) -> device:
+        """
+        `torch.device`: The device on which the module is (assuming that all the module parameters are on the same
+        device).
+        """
+        return get_parameter_device(self)
+
+    @property
+    def dtype(self) -> torch.dtype:
+        """
+        `torch.dtype`: The dtype of the module (assuming that all the module parameters have the same dtype).
+        """
+        return get_parameter_dtype(self)
+
+    def forward(
+        self,
+        sample: torch.FloatTensor,
+        timestep: Union[torch.Tensor, float, int],
+        encoder_hidden_states: torch.Tensor,
+        controlnet_conditions: List[ControlNetCondition],
+        class_labels: Optional[torch.Tensor] = None,
+        timestep_cond: Optional[torch.Tensor] = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        cross_attention_kwargs: Optional[Dict[str, Any]] = None,
+        return_dict: bool = True,
+    ) -> Union[ControlNetOutput, Tuple]:
+        if len(controlnet_conditions) != len(self.nets):
+            raise ValueError(
+                f"The number of specified `ControlNetCondition` does not match the number of `ControlNetModel`."
+                f"There are {len(self.nets)} ControlNetModel(s), "
+                f"but there are {len(controlnet_conditions)} `ControlNetCondition` in `controlnet_conditions`."
+            )
+
+        for i, (cond, controlnet) in enumerate(zip(controlnet_conditions, self.nets)):
+            down_samples, mid_sample = controlnet(
+                sample,
+                timestep,
+                encoder_hidden_states,
+                cond.image,
+                class_labels,
+                timestep_cond,
+                attention_mask,
```

Comment:
```suggestion
                attention_mask,
                cond_scale,
```
Maybe the controlnet can already take care of this:
```py
            down_samples = [sample * cond.scale for sample in down_samples]
            mid_sample *= cond.scale
```

This would allow to pass the output of the condnet directly to the unet so that this: https://github.com/huggingface/diffusers/pull/2627/files#r1132302584 can still be deleted.